instrumentation/log4j/log4j-appender-2.17/library/README.md

```diff
@@ -56,3 +56,39 @@ The following demonstrates how you might configure the appender in your `log4j.x
 
 In this example Log4j2 log events will be sent to both the console appender and
 the `OpenTelemetryAppender`.
+
+In order to function, `OpenTelemetryAppender` needs access to an `OpenTelemetry` instance. This can
+be programmatically set during application startup as follows:
+
+```java
+import io.opentelemetry.instrumentation.log4j.appender.v2_17.OpenTelemetryAppender;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+
+public class Application {
+
+  public static void main(String[] args) {
+    OpenTelemetrySdk openTelemetrySdk = // Configure OpenTelemetrySdk
+
+    // Find OpenTelemetryAppender in log4j configuration and install openTelemetrySdk
+    OpenTelemetryAppender.install(openTelemetrySdk);
+
+    // ... proceed with application
+  }
+}
+```
+
+Optionally, you can configure `OpenTelemetryAppender` to use the `GlobalOpenTelemetry` instance:
+```xml
+<OpenTelemetry name="OpenTelemetryAppender" useGlobalOpenTelemetry="true" />
+```
+
+Note: Setting `useGlobalOpenTelemetry` causes `OpenTelemetryAppender` to
+call `GlobalOpenTelemetry.get()`, which initializes `GlobalOpenTelemetry` and prevents future calls
+to `GlobalOpenTelemetry.set(...)`. `GlobalOpenTelemetry.get()` will return a noop instance unless
+you've opted
+into [autoconfigure](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/autoconfigure)
+via:
+
+- Add dependency on `io.opentelemetry:opentelemetry-sdk-extension-autoconfigure:OPENTELEMETRY_VERSION`
+- Opt into `GlobalOpenTelemetry` autoconfigure by
+  setting `-Dotel.java.global-autoconfigure.enabled=true`
```

Comment:
do we think we really need the `useGlobalOpenTelemetry` option?

Reply:
I'm happy enough to get rid of it, but that means there is no way (besides the java agent) to use an appender without some programatic configuration.

Reply:
please keep the global otel option - programmatic configuration is too much to include in every application that doesn't use the agent.

Reply:
I'm interested in better understanding the use case for using the SDK without doing anything programmatically?

Reply:
Talked about this (briefly) in the 6/29 java SIG and decided to remove the support for using GlobalOpenTelemetry. I've pushed a commit which does this.